### PR TITLE
428 update visa visibility for provider details page

### PR DESCRIPTION
**Pull request by SamJamCul**
### Context
https://trello.com/c/9xO3jEpa/428-update-visa-visibility-for-provider-details-page

### Changes proposed in this pull request
On the provider details page, providers will now be shown separate change links for each visa type - student and skilled worker. These will appear in different combinations depending on the `provider_type`.

Each change link goes through to a different form, which handles that specific visa type. 

image

image

image

### Guidance to review
Go to the organisation page for a Lead school, SCITT, and HEI. They will have different visibilities for the Visa sponsorship section. Only HEIs can make changes to either visa type. SCITTs can only change the skilled worker visa, while lead schools cannot even see the section. 

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [ ] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Review by darokel — COMMENTED** [review]
Review app looks good. But I've left some comments around the code design. Tagging @simonwhatley to also take a look.

SCITT example - https://publish-teacher-training-pr-2976.london.cloudapps.digital/publish/organisations/D87/2022/courses
HEI example - https://publish-teacher-training-pr-2976.london.cloudapps.digital/publish/organisations/B35/2022/courses

File: app/views/publish/providers/details.html.erb [hunk 1]
```diff
@@ -62,17 +62,33 @@
...
+      <h2 class="govuk-heading-m">Visa sponsorship</h2>
+      <%= govuk_summary_list do |summary_list| %>
+        <% enrichment_summary(
+             summary_list,
+             :provider,
+             "Student visa",
+             student_visa_sponsorship_status(@provider),
+             %w[can_sponsor_student_visa],
+             truncate_value: false,
+             action_path: @provider.provider_type == "university" ? student_visa_publish_provider_recruitment_cycle_path(@provider.provider_code, @provider.recruitment_cycle_year) : nil,
```
[the remark this answers is not in the document]
Comment (SamJamCul) [reply app/views/publish/providers/details.html.erb]: Easy, sorted!

File: app/controllers/publish/providers/visas_controller.rb [hunk 2]
```diff
@@ -1,35 +1,66 @@
 module Publish
   module Providers
     class VisasController < PublishController
-      def edit
+      def student_edit
```
[the remark this answers is not in the document]
Comment (SamJamCul) [reply app/controllers/publish/providers/visas_controller.rb]: Easy, split it out into separate controllers for each visa type!

**Review by goodviber — COMMENTED** [review]
Do we need the Gemfile.lock?

**Comment by SamJamCul** [comment]
> Do we need the Gemfile.lock?

How did that get there!

**Comment by defong** [comment]
[Update visa visibility for provider details page](https://trello.com/c/9xO3jEpa/428-update-visa-visibility-for-provider-details-page)

**Review by goodviber — APPROVED** [review]

**Review by darokel — APPROVED** [review]

**Comment by simonwhatley** [comment]
The summary list labels in the visa sponsorship section need to be plural - "visas":

Screenshot 2022-09-13 at 11 03 11@2x

**Comment by defong** [comment]
> The summary list labels in the visa sponsorship section need to be plural - "visas":
> 
> Screenshot 2022-09-13 at 11 03 11@2x

Fixed
image

**Pull request merged**